base/math.jl

```diff
@@ -975,6 +975,8 @@ julia> rem2pi(7pi/4, RoundDown)
 """
 function rem2pi end
 function rem2pi(x::Float64, ::RoundingMode{:Nearest})
+    isnan(x) && return NaN
+
     abs(x) < pi && return x
```

Comment:
I guess this can be optimized:
```suggestion
    abs(x) >= pi || return x
```

Reply:
Yes. but wouldn't that will make it hard to understand and debug later?

Reply:
That's right. However, for such low-level functions, I prefer the speed. Is it not enough to just add comments into the source?

Reply:
Worth checking if LLVM figures that out and eliminates the `isnan` check.

Reply:
It seems like LLVM with `optimize=true` can't eliminate the `fcmp ord double %0, 0.000000e+00` instruction and extra branch from `isnan` here. So I think it makes sense to use `>=` here to replace the `isnan` check and just add an explanatory comment about the `NaN` case.